core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/StackOperationService.java

```diff
@@ -113,54 +113,57 @@ public FlowIdentifier removeInstance(Stack stack, String instanceId, boolean for
-                    instanceIdsByHostgroupMap.computeIfAbsent(metaData.getInstanceGroupName(), s -> new LinkedHashSet<>()).add(metaData.getPrivateId());
-                }
-                updateNodeCountValidator.validateServiceRoles(stack, instanceIdsByHostgroupMap.entrySet()
-                        .stream()
-                        .collect(Collectors.toMap(Entry::getKey, e -> e.getValue().size() * -1)));
-                LOGGER.info("ZZZ: removeInstances: instanceIds/hostnames received: {}", instanceIds);
-                LOGGER.info("ZZZ: removeInstances: computedInstanceIds: {}", instanceIdsByHostgroupMap);
-                // TODO CB-14929: Validations for start-stop
-                if (!forced) {
-                    for (Entry<String, Set<Long>> entry : instanceIdsByHostgroupMap.entrySet()) {
-                        String instanceGroupName = entry.getKey();
-                        int scalingAdjustment = entry.getValue().size() * -1;
-                        updateNodeCountValidator.validateScalabilityOfInstanceGroup(stack, instanceGroupName, scalingAdjustment);
-                        updateNodeCountValidator.validateScalingAdjustment(instanceGroupName, scalingAdjustment, stack);
-                    }
-                }
-                if (scalingStrategy != null) {
-                    // TODO: CB-14929: CB-15154 stopstart can only support a single hostGroup being specified.
-                    if (instanceIdsByHostgroupMap.size() > 1) {
-                        throw new BadRequestException("Downscale via Instance Stop cannot process more than one host group");
-                    }
-                    stackUpdater.updateStackStatus(stack.getId(), DetailedStackStatus.DOWNSCALE_BY_STOP_REQUESTED,
-                            "Requested node count for downscaling (stopstart): " + instanceIds.size());
-                    return flowManager.triggerStopStartStackDownscale(stack.getId(), instanceIdsByHostgroupMap, forced);
-                } else {
-                    return flowManager.triggerStackRemoveInstances(stack.getId(), instanceIdsByHostgroupMap, forced);
-                }
-            });
-        } catch (TransactionExecutionException e) {
-            if (e.getCause() instanceof BadRequestException) {
-                throw e.getCause();
+    public FlowIdentifier stopInstances(Stack stack, Collection<String> instanceIds, boolean forced) {
+        LOGGER.info("Received stop instances request for instanceIds: [{}]", instanceIds);
+
+        if (instanceIds == null || instanceIds.isEmpty()) {
+            throw new BadRequestException("Stop request cannot process an empty instanceIds collection");
+        }
+        Map<String, Set<Long>> instanceIdsByHostgroupMap = new HashMap<>();
+        for (String instanceId : instanceIds) {
+            InstanceMetaData metaData = updateNodeCountValidator.validateInstanceForStop(instanceId, stack);
+            if (metaData != null) {
```

Comment:
If instances are not found in this step - can we collate these together and log relevant information. That will happen in the flow itself. This is more about when the particular instance disappeared.